activerecord/lib/active_record/reflection.rb

```diff
@@ -32,8 +32,11 @@ def self.create(macro, name, scope, options, ar)
     end
 
     def self.add_reflection(ar, name, reflection)
-      ar.clear_reflections_cache
-      ar._reflections = ar._reflections.merge(name.to_s => reflection)
+      # Add reflection to the current class, and downstream it to its descendants
+      (ar.descendants << ar).each do |record|
```

Comment:
Don't know how to improve on `ar.descendants << ar`, but I do have a :blush: reaaally :blush: small nitpick: `record` var name is usually used for AR instances, maybe switch `record` -> `ar_class`?